src/main/java/com/hubspot/jinjava/objects/serialization/MapEntrySerializer.java

```diff
@@ -2,28 +2,46 @@
 
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectWriter;
 import com.fasterxml.jackson.databind.SerializerProvider;
+import java.io.CharArrayWriter;
 import java.io.IOException;
-import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.atomic.AtomicInteger;
 
-public class MapEntrySerializer extends JsonSerializer<Map.Entry> {
+public class MapEntrySerializer extends JsonSerializer<Entry<?, ?>> {
   public static final MapEntrySerializer INSTANCE = new MapEntrySerializer();
 
   private MapEntrySerializer() {}
 
   @Override
   public void serialize(
-    Map.Entry object,
+    Entry<?, ?> entry,
     JsonGenerator jsonGenerator,
     SerializerProvider serializerProvider
   )
     throws IOException {
-    String key = PyishObjectMapper.PYISH_OBJECT_WRITER.writeValueAsString(
-      object.getKey()
-    );
-    String value = PyishObjectMapper.PYISH_OBJECT_WRITER.writeValueAsString(
-      object.getValue()
+    AtomicInteger remainingLength = (AtomicInteger) serializerProvider.getAttribute(
+      LengthLimitingWriter.REMAINING_LENGTH_ATTRIBUTE
     );
+    String key;
+    String value;
+    if (remainingLength != null) {
+      ObjectWriter objectWriter = PyishObjectMapper.PYISH_OBJECT_WRITER.withAttribute(
+        LengthLimitingWriter.REMAINING_LENGTH_ATTRIBUTE,
+        remainingLength
+      );
+      key = objectWriter.writeValueAsString(entry.getKey());
```

Comment:
It will count towards the limit. There is a LengthLimitingWriter already on `jsonGenerator` so `writeRawValue` will count towards the value.
Maps that are created in jinjava are `Map<String, Object>` so writing the keys of that is a single step process that won't be recursive. Writing the Objects can be recursive as the object can contain many other objects. So by creating another LengthLimitingWriter here for the `entry.value`, if the object is very expensive to serialize, we are able to break before the `entry.value` is entirely serialized. If we didn't create a new LengthLimitingWriter here, then we may have to wait for the entire `entry.value` to be serialized to check if it's too long.

I could do the same thing for the `key`, but I am trying to be as optimized as possible, and since the keys' serialization are almost never recursive, I'm not putting an additional layer of limit checking over them